term-entry: kotlin periodUntil (#3840)
Codecademy/docs: content/kotlin/concepts/datetime/terms/periodUntil/periodUntil.md

---
Title: '.periodUntil()'
Description: 'returns an object representing the difference between two dates in terms of years, months and days.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Date'
  - 'Methods'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

> The **periodUntil()** function is used to calculate the difference between two `LocalDate` or `Instant` instances. This returns an `object` representing the difference between the two dates in terms of years, months and days.

Reviewer: ```suggestion
The **`.periodUntil()`** function is used to calculate the difference between two `LocalDate` or `Instant` instances. This returns an `object` representing the difference between the two dates in terms of years, months, and days.
```

## Syntax

```pseudo
expect fun LocalDate.periodUntil(other: LocalDate): DatePeriod
```

> `DatePeriod`gives the difference between two instants, decomposed into date and time components where all time components are equals to zero.

Reviewer: Use bullet points to explain the parameters and explain the LocalDate parameter too.

> `DatePeriod`gives the difference between two instants, decomposed into date and time components where all time components are equals to zero.

Reviewer: ```suggestion
`DatePeriod` gives the difference between two instants, decomposed into date and time components where all time components are equal to zero.
```

## Example

```kotlin
val startDate = LocalDate.of(2023,11,2)
val endDate = LocalDate.of(2023,11,3)

val period = startDate.periodUntil(endDate)

> println("difference: $period.days day")

Reviewer: The code is not compiling properly, please check if the code is working or not.

Author: Hi @avdhoottt, where to add the prerequisites section? Inside the example or have to make a new section at the end?

Reviewer: Use this [guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md) and write a note describing what the user will need in order to run this code. Add the note above the code.

## Syntax

```pseudo
expect fun LocalDate.periodUntil(other: LocalDate): DatePeriod
```

- `DatePeriod` gives the difference between two instants, decomposed into date and time components where all time components are equal to zero.
- `LocalDate` represents a specific civil date without a reference to a particular time zone.

> **Note:** This method is fall under `kotlinx-datetime` library which is published to `Maven central`. Maven central repository should be added to the file first.

Maven central repository must be added to the `build.gradle.kts` file first if it is not already there,

```pseudo
repositories {
  mavenCentral()
}
```

To use the library in a single-platform project, a dependency must be added to the dependencies block.

```pseudo
dependencies {
implementation("org.jetbrains.kotlinx:kotlinx-datetime:0.4.1")
}
```

## Example

```kotlin
import java.time.LocalDate
import kotlinx.datetime.periodUntil

fun main() {
  val startDate = LocalDate.of(2023,11,2)
  val endDate = LocalDate.of(2023,11,3)

  val period = startDate.periodUntil(endDate)

  println("difference: $period.days day")
}
```

> 
> `startDate` and `endDate` given above are the `LocalDate` instances representing the start-date and the end-date.
> 

Reviewer: ```suggestion

`startDate` and `endDate` given above are the `LocalDate` instances representing the start-date and the end-date.

The result of the above code is as follows:

```

---
Title: '.periodUntil()'
Description: 'Returns an object representing the difference between two dates in terms of years, months, and days.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Date'
  - 'Methods'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

> The **`.periodUntil()`** function is used to calculate the difference between two `LocalDate` or `Instant` instances. This returns an `object` representing the difference between the two dates in terms of years, months, and days.

Reviewer 2: ```suggestion
The **`.periodUntil()`** function is used to calculate the difference between two `LocalDate` or `Instant` instances. This returns an object representing the difference between the two dates in terms of years, months, and days.
```

## Syntax

```pseudo
expect fun LocalDate.periodUntil(other: LocalDate): DatePeriod
```

- `DatePeriod` gives the difference between two instants, decomposed into date and time components where all time components are equal to zero.
- `LocalDate` represents a specific civil date without a reference to a particular time zone.

> > **Note:** This method is fall under `kotlinx-datetime` library which is published to `Maven central`. Maven central repository should be added to the file first.

Reviewer 2: ```suggestion
> **Note:** This method is part of `kotlinx-datetime` library which is published to Maven central. Maven central repository should be added to the file first.
```

> Maven central repository must be added to the `build.gradle.kts` file first if it is not already there,

Reviewer 2: ```suggestion
Maven central repository must be added to the `build.gradle.kts` file first if it is not already there;
```

```pseudo
repositories {
  mavenCentral()
}
```

To use the library in a single-platform project, a dependency must be added to the dependencies block.

```pseudo
dependencies {
implementation("org.jetbrains.kotlinx:kotlinx-datetime:0.4.1")
}
```

## Example

```kotlin
import java.time.LocalDate
import kotlinx.datetime.periodUntil

fun main() {
  val startDate = LocalDate.of(2023,11,2)
  val endDate = LocalDate.of(2023,11,3)

  val period = startDate.periodUntil(endDate)

  println("difference: $period.days day")
}
```

> `startDate` and `endDate` given above are the `LocalDate` instances representing the start-date and the end-date.

Reviewer 2: Can we rewrite so line 60 is before the code block example, introducing it?

Author: Hi @SSwiniarski, do you mean to move line 60 above ??

Reviewer 2: @vrun1208 above line 46 between `## Example` and the following code block.
